.github/workflows/canister-tests.yml

```diff
@@ -0,0 +1,292 @@
+          name: internet_identity.wasm
+          path: out/internet_identity.wasm
+
+  # A version of II where the captcha value is always the same ("a") and where
+  # the frontend is built in "development" mode (where the root key is always
+  # fetched)
+  docker-build-dummy-captcha:
+    runs-on: ubuntu-latest
+    needs: docker-build-base
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up docker buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build internet identity
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: Dockerfile
+          build-args: |
+            USE_DUMMY_CAPTCHA=1
+            II_ENV=development
+          cache-from: type=gha,scope=cached-stage
+          # Exports the artefacts from the final stage
+          outputs: ./out
+
+      - run: sha256sum out/internet_identity.wasm
+      - name: 'Upload dummy captcha wasm'
+        uses: actions/upload-artifact@v2
+        with:
+          name: internet_identity_dummy_captcha.wasm
+          path: out/internet_identity.wasm
+
+  #####################
+  # The backend tests #
+  #####################
+
+  backend:
+    runs-on: ${{ matrix.os }}
+    needs: docker-build-dummy-captcha
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, macos-latest ]
+    env:
+      GHC_VERSION: 8.8.4
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Cache ~/.cabal/store
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cabal/store
+          key: cabal-${{ runner.os }}-${{ env.GHC_VERSION }}-${{ hashFiles('backend-tests/cabal.project', 'backend-tests//cabal.project.freeze') }}
+          restore-keys: cabal-${{ runner.os }}-${{ env.GHC_VERSION }}-
+
+      - uses: actions/setup-haskell@v1.1.3
+        with:
+          ghc-version: ${{ env.GHC_VERSION }}
+          cabal-version: "3.2"
+
+      - name: Build test runner
+        working-directory: backend-tests
+        run: |
+          cabal update
+          # we cannot use `cabal install` because the build is performed in
+          # another directory which breaks the relative path used in
+          # backend-tests.hs to internet_identity.did
+          cabal build --with-ghc ghc-${{ env.GHC_VERSION }}
+
+      - name: 'Download wasm'
+        uses: actions/download-artifact@v2
+        with:
+          name: internet_identity_dummy_captcha.wasm
+          path: .
+
+      - name: Run Tests
+        shell: bash
+        working-directory: backend-tests
+        run: cabal run
+
+  ######################
+  # The selenium tests #
+  ######################
+
+  selenium:
+    runs-on: ubuntu-latest
+    needs: docker-build-dummy-captcha
+    strategy:
+      matrix:
+        start-flag: [ '', '--emulator' ]
+        device: [ 'desktop', 'mobile' ]
+      # Make sure that one failing test does not cancel all other matrix jobs
+      fail-fast: false
+    env:
+      DFX_VERSION: 0.8.3
```

Comment:
why this version? Isn't the latest 0.9.0?

Reply:
because we've been using this so far, will update in another PR!